static/src/javascripts/projects/common/modules/media/videojs-plugins/skip-ad.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+/* eslint-disable no-underscore-dangle */
+import qwery from 'qwery';
+import bean from 'bean';
+import $ from 'lib/$';
+
+/**
+    videojs plugins can't use arrow functions
+    as 'this' needs to be available as it 
+    represents an instance of the videojs player
+**/
+const skipAd = function skipAd(mediaType: string, skipTimeout: number): void {
+    let intervalId;
+
+    const skip = (): void => {
+        $('.js-ads-skip', this.el()).hide();
+        this.trigger(`${mediaType}:preroll:skip`);
+        // This is to follow more closely the videojs convention
+        this.trigger('adskip');
+        // in lieu of a 'skip' api, rather hacky way of achieving it
+        this.ima.onAdComplete_();
+        this.ima.onContentResumeRequested_();
+        this.ima.getAdsManager().stop();
+    };
+
+    const update = (): void => {
+        const adsManager = this.ima.getAdsManager();
+        const currentTime =
+            adsManager.getCurrentAd().getDuration() -
+            adsManager.getRemainingTime();
+        const skipTime = parseInt((skipTimeout - currentTime).toFixed(), 10);
+
+        if (skipTime > 0) {
+            $('.js-skip-remaining-time', this.el()).text(skipTime);
+        } else {
+            window.clearInterval(intervalId);
+            $('.js-ads-skip', this.el()).html(
+                `<button class="js-ads-skip-button vjs-ads-skip__button" data-link-name="Skip video advert">
+                    <i class="i i-play-icon-grey skip-icon"></i>
+                    <i class="i i-play-icon-gold skip-icon"></i>Skip advert
+                </button>`
+            );
+            bean.on(qwery('.js-ads-skip-button')[0], 'click', skip);
```

Comment:
Could be even `document.querySelector` since it's accessing the DomNode with`[0]`.